src/System.Management.Automation/namespaces/FileSystemProvider.cs

```diff
@@ -6666,7 +6664,11 @@ public IContentReader GetContentReader(string path)
 
             try
             {
-                if (Directory.Exists(path))
+                if (Directory.Exists(path)
+#if !UNIX
+                    && string.IsNullOrEmpty(streamName)
```

Comment:
It would be nice to add a comment about the behavior like "On Windows Get-Content -Stream '' will write non-terminating error"

Reply:
On Windows, `Get-Content directory -Stream streamname` will act like a data stream on a file.  On non-Windows, data stream semantics simply don't exist, so we don't have to run through the processing of whether a stream is being acted on.

Reply:
Comment added.